src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -209,48 +209,41 @@ private Resolved resolveIndexPatterns(final IndicesOptions indicesOptions, final
         }
 
         final Set<String> matchingAliases;
-        final Set<String> matchingIndices;
         final Set<String> matchingAllIndices;
 
         if (isLocalAll(requestedPatterns0)) {
             if (log.isTraceEnabled()) {
                 log.trace(Arrays.toString(requestedPatterns0) + " is an LOCAL ALL pattern");
             }
             matchingAliases = Resolved.All_SET;
-            matchingIndices = Resolved.All_SET;
             matchingAllIndices = Resolved.All_SET;
 
         } else if (!remoteIndices.isEmpty() && localRequestedPatterns.isEmpty()) {
             if (log.isTraceEnabled()) {
                 log.trace(Arrays.toString(requestedPatterns0) + " is an LOCAL EMPTY request");
             }
-            return new Resolved.Builder().addOriginalRequested(Arrays.asList(requestedPatterns0)).addRemoteIndices(remoteIndices).build();
+            return new Resolved(Collections.emptySet(), Collections.emptySet(), Collections.emptySet(), Sets.newHashSet(requestedPatterns0), remoteIndices);
         }
 
         else {
 
             ClusterState state = clusterService.state();
-
+            Set<String> dateResolvedLocalRequestedPatterns = localRequestedPatterns
+                            .stream()
+                            .map(resolver::resolveDateMathExpression)
+                            .collect(Collectors.toSet());
+            //fill matchingAliases
             final SortedMap<String, AliasOrIndex> lookup = state.metaData().getAliasAndIndexLookup();
-            final Set<String> aliases = lookup.entrySet().stream().filter(e -> e.getValue().isAlias()).map(e -> e.getKey())
+            matchingAliases = lookup.entrySet()
+                    .stream()
+                    .filter(e -> e.getValue().isAlias())
+                    .map(Map.Entry::getKey)
+                    .filter(alias -> WildcardMatcher.matchAny(dateResolvedLocalRequestedPatterns, alias))
                     .collect(Collectors.toSet());
 
-            matchingAliases = new HashSet<>(localRequestedPatterns.size() * 10);
-            matchingIndices = new HashSet<>(localRequestedPatterns.size() * 10);
-            matchingAllIndices = new HashSet<>(localRequestedPatterns.size() * 10);
-
-            //fill matchingAliases
-            for (String localRequestedPattern : localRequestedPatterns) {
-                final String requestedPattern = resolver.resolveDateMathExpression(localRequestedPattern);
-                final List<String> _aliases = WildcardMatcher.getMatchAny(requestedPattern, aliases);
-                matchingAliases.addAll(_aliases);
-            }
-
-
-            List<String> _indices;
+            Set<String> _indices;
```

Comment:
nit: declaring `matchingAllIndices` `final` does not add any value and makes `_indices` necessary. Drop `final` from `matchingAliases` and `matchingAllIndices` and remove `_indices` local variable.

Reply:
done